From 7915ad79cd3e231d2b318e47412f15e19534cba8 Mon Sep 17 00:00:00 2001
From: Fanrui Meng <mengfanrui@loongson.cn>
Date: Wed, 25 May 2022 11:17:05 +0800
Subject: [PATCH 47/48] LoongArch32r : modify la32r compile toolchain config.

	1. la32r toolchain config modification
	2. la32r libc need futex(TIME32_SYSCALL)

Signed-off-by: Fanrui Meng <mengfanrui@loongson.cn>
---
 arch/loongarch/Makefile                  | 6 +++---
 arch/loongarch/configs/la32_defconfig    | 2 +-
 arch/loongarch/include/uapi/asm/unistd.h | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/loongarch/Makefile b/arch/loongarch/Makefile
index af02419b7bcc..2910ab6cf0e6 100644
--- a/arch/loongarch/Makefile
+++ b/arch/loongarch/Makefile
@@ -10,7 +10,7 @@ KBUILD_DTBS      := dtbs
 #
 # Select the object file format to substitute into the linker script.
 #
-32bit-tool-archpref     = loongarch32
+32bit-tool-archpref     = loongarch32r
 64bit-tool-archpref	= loongarch64
 32bit-bfd		= elf32-loongarch
 64bit-bfd		= elf64-loongarch
@@ -18,7 +18,7 @@ KBUILD_DTBS      := dtbs
 64bit-emul		= elf64loongarch
 ifdef CONFIG_32BIT
 tool-archpref           =$(32bit-tool-archpref)
-UTS_MACHINE             := loongarch32
+UTS_MACHINE             := loongarch32r
 endif
 
 ifdef CONFIG_64BIT
@@ -36,7 +36,7 @@ ifdef CONFIG_32BIT
 ld-emul                 = $(32bit-emul)
 vmlinux-32              = vmlinux.32
 
-cflags-y                += -mabi=ilp32
+cflags-y                += -mabi=ilp32s
 endif
 
 
diff --git a/arch/loongarch/configs/la32_defconfig b/arch/loongarch/configs/la32_defconfig
index a56b31dff1bb..aec111efbf5c 100644
--- a/arch/loongarch/configs/la32_defconfig
+++ b/arch/loongarch/configs/la32_defconfig
@@ -2,7 +2,7 @@
 # Automatically generated file; DO NOT EDIT.
 # Linux/loongarch 5.14.0-rc2 Kernel Configuration
 #
-CONFIG_CC_VERSION_TEXT="loongarch32-linux-gnu-gcc (GCC) 8.3.0"
+CONFIG_CC_VERSION_TEXT="loongarch32r-linux-gnusf-gcc (GCC) 8.3.0"
 CONFIG_CC_IS_GCC=y
 CONFIG_GCC_VERSION=80300
 CONFIG_CLANG_VERSION=0
diff --git a/arch/loongarch/include/uapi/asm/unistd.h b/arch/loongarch/include/uapi/asm/unistd.h
index 5797898a43c7..362f6c714c91 100644
--- a/arch/loongarch/include/uapi/asm/unistd.h
+++ b/arch/loongarch/include/uapi/asm/unistd.h
@@ -4,7 +4,7 @@
 #define __ARCH_WANT_SYS_CLONE3
 #define __ARCH_WANT_SET_GET_RLIMIT
 
-#ifdef __loongarch32
+#ifdef __loongarch32r
 #define __ARCH_WANT_STAT64
 #define __ARCH_WANT_TIME32_SYSCALLS
 #endif
-- 
2.37.2

